Assets/_Scripts/BattleStateListener.cs

```diff
@@ -0,0 +1,80 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+// TODO: May want to store these listeners with the Game Manager
```

Comment:
No, it should be a monobehaviour in the scene because it's specific to the scene.